test/integration/exchangeIssuanceZeroExIntegration.spec.ts

```diff
@@ -0,0 +1,442 @@
+import "module-alias/register";
+import { Address, Account } from "@utils/types";
+import { ADDRESS_ZERO, MAX_UINT_256 } from "@utils/constants";
+import { SetToken } from "@utils/contracts/setV2";
+import { ether, getAccounts, getSetFixture, getWaffleExpect } from "@utils/index";
+import DeployHelper from "@utils/deploys";
+import { UnitsUtils } from "@utils/common/unitsUtils";
+import { SetFixture } from "@utils/fixtures";
+import { BigNumber, ContractTransaction } from "ethers";
+import { ExchangeIssuanceZeroEx, StandardTokenMock, WETH9 } from "@utils/contracts/index";
+import axios from "axios";
+import qs from "qs";
+import hre, { ethers } from "hardhat";
+
+const expect = getWaffleExpect();
+
+type ZeroExSwapQuote = {
+  sellToken: Address;
+  buyToken: Address;
+  swapCallData: string;
+};
+
+type SetTokenScenario = {
+  setToken: Address;
+  controller: Address;
+  issuanceModuleAddress: Address;
+  isDebtIssuanceModule: boolean;
+};
+
+type TokenName = "SimpleToken" | "DPI";
+
+function logVerbose(...args: any[]) {
+  if (process.env.VERBOSE) console.log(...args);
+}
+
+if (process.env.INTEGRATIONTEST) {
```

Comment:
I added this flag to make extra sure that these tests are not run in the normal test / ci pipeline. Not sure if actually necessary

Reply:
The reasons why I didn't want them to be run in the normal ci:
1. The local chain they are run against is different (forking from the latest block)
2. The tests are not necessarily reproducible since they depend on the 0x api response, which is outside of our control.

Point 2.  could be changed by generating / saving a bunch of 0x api quotes for a given block and then run the tests against that fixed block number with those quotes. 

However the point of these integration tests for me was to simulate a workflow as close as possible of how it would be used in production, and manually play with the parameters.